lib/IAMClient.js

```diff
@@ -316,23 +360,22 @@ class IAMClient {
         }
 
         let ret = '';
+        if (method === 'GET') {
+            if (reqUid) {
+                options.headers = {
+                    'x-scal-request-uids': reqUid,
+                };
+            }
+            options.path += `?${queryString.stringify(data)}`;
+        }
         const req = this.useHttps ?
             https.request(options) : http.request(options);
-
-        if (method === 'POST' && typeof data === 'object') {
+        if (method === 'POST') {
             auth.generateV4Headers(req, data,
                     this.accessKey, this.secretKeyValue, 'iam');
             req.write(queryString.stringify(data, null, null, {
                 encodeURIComponent,
```

Comment:
Why do you specify encodeURIComponent here?  I read the node docs as saying this is for changing the default.
https://nodejs.org/api/querystring.html#querystring_querystring_stringify_obj_sep_eq_options